systemservices/marketplace/src/contracts/manifest.ts

```diff
@@ -21,12 +26,18 @@ const getManifest = async (protocol: string, source: string): Promise<Manifest|u
       console.warn('protocol', protocol, 'is not compatible with this service')
       return
     }
-    const manifest = await get[protocol](source)
-    if (typeof manifest === 'object') {
-      return manifest as Manifest
+    const manifest: Manifest = await get[protocol](source)
+    const validation = validator.validate(manifest, manifestSchema)
+    if (!validation.valid) {
+      console.warn('manifest', protocol, '::', source, 'is not a valid manifest')
```

Comment:
can we throw error in this function and catch it if we need to when we call this function. Just log in the console and return nothing is not really safe...